allen-cell-animated/simularium-website
# Fix/floating point time errors (update viewer to v2.9.2)

Problem
=======
* simularium-viewer 2.9.0 (current) has floating point precision errors that cause time- and playback-related bugs.
* simularium-website itself has a not-very-robust fix for floating point errors (arbitrary rounding of time values to 4 sig figs).

Resolves: [simularium-website - replace arbitrary time value rounding to use epsilon instead](https://aicsjira.corp.alleninstitute.org/browse/AGENTVIZ-1430)

Solution
========
* Updated simularium-viewer from 2.9.0 to 2.9.2, which contains floating point error fixes using a new `floatTimes()` function
* Used the `floatTimes()` function from simularium-viewer to deal with potential floating point issues in `ViewerPanel` as well

## Type of change
Please delete options that are not relevant.


Steps to Verify:
----------------
1. Pull this branch and `npm i`, `npm start`
2. Load up networked trajectories and see if you can reproduce [this bug](https://aicsjira.corp.alleninstitute.org/browse/AGENTVIZ-1428) or any other new bugs
